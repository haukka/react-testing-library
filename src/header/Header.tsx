import React from 'react';

interface IHeader {
    text: string
}

const Header = ({ text }: IHeader) => {
    return (
        <header>
            <h1 data-testid="tag" className="fancy-h1">
                {text}
            </h1>
        </header>
    );
};

export default Header;