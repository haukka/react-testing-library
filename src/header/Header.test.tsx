import React from "react";
import { screen, render, cleanup } from "@testing-library/react";
import Header from './Header';

afterEach(cleanup);

describe('testing Header Component', () => {
    it('test render', () => {
        const { asFragment } = render(<Header text={'test'} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('inserts text in h1', () => {
        render(<Header text={'test'} />)
        expect(screen.getByTestId('tag')).toHaveTextContent('test');
        expect(screen.getByText('test')).toHaveClass('fancy-h1');
    });
});