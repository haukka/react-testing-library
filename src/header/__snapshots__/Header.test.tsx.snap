// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header Component test render 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="fancy-h1"
      data-testid="tag"
    >
      test
    </h1>
  </header>
</DocumentFragment>
`;
